pkg/gameserverallocations/allocation_cache_test.go

```diff
@@ -80,26 +80,6 @@ func TestAllocationCacheListSortedGameServers(t *testing.T) {
 		features string
 		gsa      *allocationv1.GameServerAllocation
 	}{
-		"most allocated": {
-			// node1: 1 ready, 1 allocated, node2: 1 ready
-			list: []agonesv1.GameServer{gs1, gs2, gs3},
-			test: func(t *testing.T, list []*agonesv1.GameServer) {
-				assert.Len(t, list, 2)
-				if !assert.Equal(t, []*agonesv1.GameServer{&gs1, &gs2}, list) {
-					for _, gs := range list {
-						logrus.WithField("name", gs.Name).Info("game server")
-					}
-				}
-			},
-			features: fmt.Sprintf("%s=false", runtime.FeatureStateAllocationFilter),
-		},
-		"allocated first (StateAllocationFilter)": {
```

Comment:
I think we should have this test, since the state filter on this one is `true`.

Reply:
This test has been added.